datapackage-tabular.md

```diff
@@ -0,0 +1,1078 @@
+# `vega-datasets`- `description` Common repository for example datasets used by Vega related projects.
+- `homepage` http://github.com/vega/vega-datasets.git
+- `licenses`
+  - [1]
+    - `name` BSD-3-Clause
+    - `path` https://opensource.org/license/bsd-3-clause
+    - `title` The 3-Clause BSD License
+- `contributors`
+  - [1]
+    - `title` UW Interactive Data Lab
+    - `path` http://idl.cs.washington.edu
+- `version` 2.11.0
+- `created` 2024-12-13T12:53:03.887410+00:00
+## `7zip.png`
+  - `description` Application icons from open-source software projects.
+  - `path` 7zip.png
+## `airports.csv`
+  - `path` airports.csv
+  - `schema`
+      
+  | name      | type   |
+|:----------|:-------|
+| iata      | string |
+| name      | string |
+| city      | string |
+| state     | string |
+| country   | string |
+| latitude  | number |
+| longitude | number |
+## `annual-precip.json`
+  - `description` A raster grid of global annual precipitation for the year 2016 at a resolution 1 degree of lon/lat per cell.
+  - `path` annual-precip.json
```

Comment:
> [!WARNING]
> Both markdown outputs seems to omit `[[resources.sources]]` and `[[resources.licenses]]`

Reply:
Only just discovered this now.

Please don't merge yet

Reply:
Seems like we'll need to override this template:

https://github.com/frictionlessdata/frictionless-py/blob/6b72909ee38403df7c0245f408f3881bfa56ad6f/frictionless/assets/templates/resource-table.md

Reply:
I think this will work as the new template

<details><summary><code>resource-table.md</code></summary>
<p>

```md
## `{{ resource.name }}`{% if resource.title %} {{ resource.title }}{% endif %}

{% if resource.description %}
  - `description` {{ resource.description | indent(4, False) }}
{% endif %}
{% if resource.path %}
  - `path` {{ resource.path }}
{% endif %}
{% if resource.schema %}
  - `schema`
  {{ resource.schema | filter_dict(exclude=['fields']) | dict_to_markdown(level=2) }}
  {{ resource.schema.fields | tabulate() }}
{% endif %}
{% if resource.sources %}
  - `sources`
  {{ resource.sources | tabulate() }}
{% endif %}
{% if resource.licenses %}
  - `licenses`
  {{ resource.licenses | tabulate() }}
{% endif %}

```

</p>
</details> 

Patching this in [here](https://github.com/frictionlessdata/frictionless-py/blob/6b72909ee38403df7c0245f408f3881bfa56ad6f/frictionless/formats/markdown/mapper.py#L13-L43) seems like it'll be tricky:

<details><summary><code>frictionless.formats.markdown.mapper</code></summary>
<p>

```py
class MarkdownMapper(Mapper):
    """Markdown mapper"""

    def write_metadata(self, metadata: Metadata, *, table: bool = False):
        filename = metadata.metadata_type
        template = f"{filename}-table.md" if table is True else f"{filename}.md" # <---- File name set here
        descriptor = metadata.to_descriptor()
        text = render_markdown(f"{template}", {filename: descriptor}).strip()
        return text


# Internal


def render_markdown(path: str, data: Dict[str, Any]) -> str:
    """Render any JSON-like object as Markdown, using jinja2 template"""

    # Create environ
    template_dir = os.path.join(os.path.dirname(__file__), "../../assets/templates") # <---- Replacing file here
    environ = platform.jinja2.Environment(
        loader=platform.jinja2.FileSystemLoader(template_dir),
        lstrip_blocks=True,
        trim_blocks=True,
    )

    # Render data
    environ.filters["filter_dict"] = filter_dict  # type: ignore
    environ.filters["dict_to_markdown"] = dict_to_markdown  # type: ignore
    environ.filters["tabulate"] = dicts_to_markdown_table  # type: ignore
    template = environ.get_template(path)
    return template.render(**data)


```

</p>
</details>

Reply:
Fixed in (https://github.com/vega/vega-datasets/pull/643#commits-pushed-51fb7fc)